provider/azure/environ.go

```diff
@@ -176,15 +180,19 @@ func createStorageAccount(
 	for remaining := maxAttempts; remaining > 0; remaining-- {
 		accountName := accountNameGenerator()
 		logger.Debugf("- checking storage account name %q", accountName)
-		result, err := client.CheckNameAvailability(
-			storage.AccountCheckNameAvailabilityParameters{
-				Name: to.StringPtr(accountName),
-				// Azure is a little inconsistent with when Type is
-				// required. It's required here.
-				Type: to.StringPtr("Microsoft.Storage/storageAccounts"),
-			},
-		)
-		if err != nil {
+		var result storage.CheckNameAvailabilityResult
+		if err := callAPI(func() (autorest.Response, error) {
```

Comment:
Wouldn't help. You can't use a named function type in a function literal like that.